File: Examples/DotNet/CSharp/Workbook/ConvertWorkbookToAnotherFormatWithoutStorage.cs
using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;

namespace Workbook
{
    class ConvertWorkbookToAnotherFormatWithoutStorage
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "Sample_Test_Book";
            String inputfileName = fileName + ".xls";
            String format = "pdf";
            String password = "";
            String outPath = "";
            String outputFileName = fileName + "." + format;

            byte[] file = System.IO.File.ReadAllBytes(Common.GetDataDir() + inputfileName);

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(inputfileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + inputfileName));

                // Invoke Aspose.Cells Cloud SDK API to convert excel workbook to different format without storage
                ResponseMessage apiResponse = cellsApi.PutConvertWorkBook(format, password, outPath, file);

                if (apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("File Converted successfully to:" + outputFileName);
                    Console.ReadKey();
                }
                else
                {
                    Console.WriteLine("status:" + apiResponse.Status);
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Perl/Workbook/ConvertWorkbookToAnotherFormatWithoutStorage.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeCellsCloud::CellsApi;
use AsposeCellsCloud::ApiClient;
use AsposeCellsCloud::Configuration;
use AsposeCellsCloud::Object::SaveOptions;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeCellsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeCellsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeCellsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Cells API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $cellsApi = AsposeCellsCloud::CellsApi->new();

# Set input file name
my $filename = "Sample_Test_Book";
my $name = $filename .  ".xls";
my $format = "pdf";

# Invoke Aspose.Cells Cloud SDK API to convert workbook to different file formats without storage              		                           
my $response = $cellsApi->PutConvertWorkBook(file => $data_path.$name, format=>$format);

if($response->{'Status'} eq 'OK'){
	# Download Workbook from response stream	 
	my $output_file = $out_path. $filename . "." . $format;
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1